lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1065,6 +1117,22 @@ def exec_module(self, **kwargs):
 
                         vm_resource.storage_profile.data_disks = data_disks
 
+                    # Before creating VM accept terms of plan if `accept_terms` is True
+                    if self.accept_terms:
+                        if not self.plan:
+                            self.fail("parameter error: plan must be specified and include name, product, and publisher")
+                        try:
+                            plan_name = self.plan.get('name')
+                            plan_product = self.plan.get('product')
+                            plan_publisher = self.plan.get('publisher')
+                            term = self.marketplace_client.marketplace_agreements.get(
+                                plan_publisher, plan_product, plan_name)
+                            term.accepted = True
+                            agreement = self.marketplace_client.marketplace_agreements.create(
+                                plan_publisher, plan_product, plan_name, term)
```

Comment:
Need these to be kwargs for safety; generated Azure SDK is unsafe to use positional args. We're trying to move everything there to use kwargs.